### PR TITLE
pybind/rbd: support open the image by image_id

### PR DESCRIPTION
**Pull request by Songweibin**
We can initialize an image with image name or image id.

Signed-off-by: songweibin <song.weibin@zte.com.cn>

### CODE REVIEW

File: src/pybind/rbd/rbd.pyx [hunk 1]
```diff
@@ -1360,9 +1364,11 @@ cdef class Image(object):
     cdef object ioctx
     cdef rados_ioctx_t _ioctx
 
-    def __init__(self, ioctx, name, snapshot=None, read_only=False):
+    def __init__(self, ioctx, name=None, snapshot=None, read_only=False, img_id=None):
         """
         Open the image at the given snapshot.
+        If image name and image id are both specified, utilize image name.
```
Comment (dillaman) [inline src/pybind/rbd/rbd.pyx]: Nit: I would just throw an exception since it shouldn't be possible

**Comment by Songweibin** [comment]
@dillaman   Updated, please help to have a look, thank you.

**Comment by dillaman** [comment]
retest this please

**Comment by dillaman** [comment]
Fails Python tests:
http://qa-proxy.ceph.com/teuthology/jdillaman-2017-12-07_19:04:33-rbd-wip-jd-testing-distro-basic-smithi/1943616/teuthology.log
http://qa-proxy.ceph.com/teuthology/jdillaman-2017-12-07_19:04:33-rbd-wip-jd-testing-distro-basic-smithi/1943623/teuthology.log
http://qa-proxy.ceph.com/teuthology/jdillaman-2017-12-07_19:04:33-rbd-wip-jd-testing-distro-basic-smithi/1943630/teuthology.log

File: src/test/pybind/test_rbd.py [hunk 2]
```diff
@@ -862,6 +862,34 @@ def test_watchers_list(self):
...
+class TestImageId(object):
+
+    def setUp(self):
+        self.rbd = RBD()
+        create_image()
+        self.image = Image(ioctx, image_name)
+        self.image2 = Image(ioctx, None, None, False, self.image.id())
+
+    def tearDown(self):
+        self.image2.close()
```
Comment (dillaman) [inline src/test/pybind/test_rbd.py]: Need to close ```self.image```

**Comment by Songweibin** [comment]
@dillaman  Done. Sorry for that mistake,  thanks.

**Pull request merged**